### PR TITLE
github: automation: fixed rules for commit scope check

### PR DESCRIPTION
**Pull request by vpirogov**
This PR updates commit checker to avoid false positives in #2279 ([job link](https://github.com/oneapi-src/oneDNN/actions/runs/12393533080/job/34594975133)), #2276, and #2272.

As `:` can be used in commit message itself there's no good way to reliably identify cases with multi-level scope.

### CODE REVIEW

File: .github/automation/commit-msg-check.py [hunk 1]
```diff
@@ -20,29 +20,17 @@
...
-    # The last element of the split is the title, which we don't care about.
-    scopesArray = firstLine.split(":")[:-1]
-
-    for scopes in scopesArray:
-        numWords = len(scopes.split())
-        numCommas = scopes.count(",")
-
-        if numWords != numCommas + 1:
-            print(f"{status} FAILED: Same-level scopes must be comma-separated. Bad token: '{scopes}'")
-            return False
```
Comment (rjoursler) [inline .github/automation/commit-msg-check.py:44]: I don't see any issue with keeping this check for the top level scope. At minimum, the comment on line 25 should be updated if this is removed.

File: .github/automation/commit-msg-check.py [hunk 2]
```diff
@@ -20,29 +20,17 @@
 
 import argparse
 import subprocess
+import re
 
 # * Ensuring the scopes end in colon and same level scopes are comma delimited.
 # TODO: Limit scopes to an acceptable list of tags.
```
Comment (rjoursler) [inline .github/automation/commit-msg-check.py:26]: Please add to this comment that the current implementation can only validate the top level scope due to ambiguous parsing of the `:` token between the scope title and the description.

File: .github/automation/commit-msg-check.py [hunk 3]
```diff
@@ -79,7 +67,7 @@ def main():
       is_ok = is_ok and result
 
     if is_ok:
```
Comment (theComputeKid) [inline .github/automation/commit-msg-check.py:69]: Casually switching to snake_case halfway through the file that started with camelCase..

File: .github/automation/commit-msg-check.py (hunk 1 above)
Comment (vpirogov) [reply .github/automation/commit-msg-check.py:44]: Right. Technically coding guidelines do not allow comma for top level scope, though I see that we do that from time to time in the commit history.

**Comment by theComputeKid** [comment]
This means that e.g. `Removing: from doc` would be a valid commit title, since there is no validation of the initial tag.

File: .github/automation/commit-msg-check.py (hunk 3 above)
Comment (vpirogov) [reply .github/automation/commit-msg-check.py:69]: Looks like argparse package set the tone for `main()`...

File: .github/automation/commit-msg-check.py [hunk 4]
```diff
@@ -20,33 +20,23 @@
 
 import argparse
 import subprocess
+import re
 
-# * Ensuring the scopes end in colon and same level scopes are comma delimited.
+# Ensuring the scope ends in colon and same level scopes are comma delimited.
```
Comment (rjoursler) [inline .github/automation/commit-msg-check.py:25]:
```Suggestion
# Ensure the scope ends in a colon and that same level scopes are comma delimited.
```

**Review by rjoursler — APPROVED** [review]

**Comment by vpirogov** [comment]
> This means that e.g. `Removing: from doc` would be a valid commit title, since there is no validation of the initial tag.

No, this one will fails because of capital 'R' :). Anything that follows the format of `<scope>: <description>` or `<scope1>, <scope2>: <description>` will fly. Testing on commits since oneDNN v3.5 release indicates that nothing really ugly passes and less than 3% of commits fail.

**Review by theComputeKid — APPROVED** [review]

File: .github/automation/commit-msg-check.py [hunk 5]
```diff
@@ -67,7 +58,8 @@ def main():
     head: str = args.head
 
     commit_range = base + ".." + head
-    messages = subprocess.run(["git", "rev-list", "--format=oneline", commit_range], capture_output=True, text=True).stdout
+    messages = subprocess.run(["git", "rev-list", "--format=oneline",
+               commit_range], capture_output=True, text=True).stdout
```
Comment (mgouicem) [inline .github/automation/commit-msg-check.py:62]: (nit) according to [peps8](https://peps.python.org/pep-0008/#indentation), arguments should 
- either start on the next line with 4 spaces offset, 
- or start on same line but upon line break be aligned with the opening bracket

**Review by mgouicem — APPROVED** [review]

File: .github/automation/commit-msg-check.py (hunk 5 above)
Comment (vpirogov) [reply .github/automation/commit-msg-check.py:62]: we need clang-format but for Python...

**Pull request merged**